File: solutions/zig/raindrops/2/raindrops.zig
const std = @import("std");

pub fn convert(buffer: []u8, n: u32) []const u8 {
    const sounds = std.fmt.bufPrint(buffer, "{s}{s}{s}", .{
        if (n % 3 == 0) "Pling" else "",
        if (n % 5 == 0) "Plang" else "",
        if (n % 7 == 0) "Plong" else "",
    }) catch unreachable;

    if (sounds.len > 0) {
        return sounds;
    } else {
        return std.fmt.bufPrint(buffer, "{d}", .{n}) catch unreachable;
    }
}

File: solutions/go/raindrops/3/raindrops.go
// Package raindrops converts rain to sounds
package raindrops

import (
	"strconv"
)

// Convert produces a string of sounds a number would make.
func Convert(number int) string {
	var drops string

	if number%3 == 0 {
		drops += "Pling"
	}
	if number%5 == 0 {
		drops += "Plang"
	}
	if number%7 == 0 {
		drops += "Plong"
	}
	if drops == "" {
		return strconv.Itoa(number)
	}

	return drops
}
